Generate valid SARIF files

indirect opened the pull request:
According to the [official SARIF validator](https://sarifweb.azurewebsites.net/Validation), rules need to provide a `helpUri` property. This adds help URLs for all the cops in departments hosted at docs.rubocop.org, and tweaks a couple other minor things so that the generated JSON passes the SARIF validator.

indirect committed:
including an index fails validation

if an index is present, the spec requires that there be another array with an item at this corresponding index, which there is not

indirect committed:
add RuboCop version and informationUri

both are required for the SARIF to be valid

indirect committed:
transform name to pass validation

names must be PascalCaseStrings, with no slashes and no double-capitals

indirect committed:
provide helpUri for all departments at rubocop.org

the SARIF validator requires a property named `helpUri`, containing a link to the docs for the rule. this uses RuboCop's built-in documentation URL, falling back on hardcoded strings for the RuboCop addon gems that are hosted at docs.rubocop.org.

indirect committed:
update gems

indirect committed:
remove empty partialFingerprints so GitHub generates one

indirect committed:
anchors don't have slashes in them

arthurnn commented on lib/code_scanning/rubocop/rule.rb:
```diff
@@ -80,19 +104,19 @@ def to_h
-          "text" => help,
-          "markdown" => help
-        }
-        properties["queryURI"] = query_uri if badge.qualified?
```
Any particular reason you removed the 'queryURI' property?

indirect replied:
Yes, I had to remove it because the official SARIF validator fails the feed if queryURI is present. :/

arthurnn replied:
@indirect the code-scanning security tab and views rely on the `queryURI` to display some information. Also, we return it in our APIs. It seems like the SARIF validator does not recognize this field, but we have it in CodeQL and the rubocop generator too in order to provide the full experience. 

I am ok with adding the `helpUri`. but we can't remove the other, at least for now.

indirect replied:
Huh, I see. Okay, sounds like perhaps CodeQL ought to accept `helpUri` instead of `queryURI` to fully conform to SARIF? I'll add it back, though. My main goal with this PR was to get the Code Scanning Alerts tab to link to the RuboCop docs. 😅

indirect replied:
Done, added queryURI back.

indirect committed:
add back queryURI for CodeQL

even though it's not technically valid SARIF, GitHub's CodeQL system
uses queryURI, so we'll provide it for now.

arthurnn reviewed (approved):
👏
